src/components/external-events/ExternalEventForm.svelte

```diff
@@ -0,0 +1,130 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { field } from '../../stores/form';
+  import { plugins } from '../../stores/plugins';
+  import type { ExternalEventDB } from '../../types/external-event';
+  import type { ExternalEventProperty } from '../../types/external-event-property';
+  import { formatDate } from '../../utilities/time';
+  import Collapse from '../Collapse.svelte';
+  import DatePickerField from '../form/DatePickerField.svelte';
+  import Input from '../form/Input.svelte';
+  import Properties from './Properties.svelte';
+
+  export let externalEvent: ExternalEventDB;
```

Comment:
`ExternalEventForm` & `ExternalEventsTable` use `ExternalEventDB` instead of `ExternalEvent` because the forms they show are meant to more-or-less be a visual representation of the event at the database level. Additionally for us the `ExternalEvent` type only adds two fields (`duration_ms` and `start_ms`) which are _only_ used in the context of `Timeline`s. I think it makes sense to keep using the `*DB` type here, but let me know if you think it would be more clear to use `ExternalEvent`.

Also, with the [last commit](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/fc56338a1e9b4c60e2ab1b1203310a84de0afc49) I cleaned up one use of `ExternalEventDB` where `Timeline` was creating the `ExternalEvent`-typed events on the fly - now they're generated as a derived store so there's less direct access to `ExternalEventsDB`-typed events!

Reply:
Yeah that makes sense. What I want to facilitate is type choosing for devs. If a dev writes a new component for external events and they try to autocomplete, they'll be presented with `ExternalEvent` and `ExternalEventDB` types. Ideally they could just rely on the convention that the type sharing the same namesake as the concept name (i.e. ExternalEvent or ActivityDirective) should have everything needed for the UI. Conventionally, the `*DB` version is missing some UI specific elements or has different fields that are specific to dealing with the db. 

This separation also helps isolate the UI components from database changes. Rather than having to update every piece of code that uses the updated *DB type, a translation could potentially be added between the *DB type and the namesake type to minimize the the number of changes overall if it makes sense to do so.

That being said, this is more just about maintainability than it is functionality. If you feel strongly about using the *DB type, I'm open to discussing it more in a huddle or something.

Reply:
I would say that the `*DB` version here too is missing UI specific elements (specifically those `duration_ms` and `start_ms` fields) which are draw specific, so I'm not sure if/how this breaks the convention. I am also open to discussing this in a huddle!

Reply:
This was resolved in a huddle. We now have `ExternalEventDB` acting as a type _only_ facing the database, instead using `ExternalEvent` anywhere else. We initially operated under the assumption that the extra, timeline-specific information in `ExternalEvent` was excessive and we should instead use `ExternalEventDB`, but this was ultimately ruled against as using `ExternalEvent` generally, in spite of its (very few) additional fields, is ultimately more intuitive for developers.